src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -46,87 +45,166 @@ class MatchSamplingRate(Transform):
 
     The resulting Dataframe after the transform is:
 
-    | time_feature_a | feature_a   | time_feature_b | feature_b    | constant |
-    | -------------- | ----------- | -------------- | ------------ | -------- |
-    | [0, 1, 2]      | [2, 1, 7]   | [0, 1, 2]       | [10, 15, 20]    | 1    |
-    | [0, 1, 2]      | [4, 1, 0]   | [0, 1, 2]       | [20, 30, 40]    | 2    |
+    | feature_a                   | feature_b                   | const |
+    | ---                         | ---                         | ---   |
+    | list[struct[time,struct[]]] | list[struct[time,struct[]]] | int   |
+    | --------------------------- | --------------------------- | ----- |
+    | [{12:26:00.0, {1.2}},       | [{12:26:00.0, {1.2}},       | 1     |
+    |  {12:26:01.0, {2.4}},       |  {12:26:01.0, {1.4}},       |       |
+    |  {12:26:02.0, {3.6}},       |  {12:26:02.0, {1.6}},       |       |
+    |  {12:26:03.0, {4.8}}]       |  {12:26:03.0, {1.8}}]       |       |
 
-
-
-    Note that the used feature `time_feature_b` is still
-    present in the DataFrame. To remove it use the `select` transform.
-    """  # noqa: E501
+    """
 
     def __init__(
         self,
-        reference_timestamps: str,
-        feature_columns_with_timestamps: dict[str, str],
+        reference_feature_name: str,
+        feature_interpolation_map: dict[str, str],
```

Comment:
Replace `str` with an `InterpolationMethodType` to constraint the choice here. This should make type checking easier and provide better hints.